driver/mloPdistHost.hpp

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include <math.h>
+
+#include <miopen/tensor.hpp>
+#include <miopen/tensor_view_utils.hpp>
+#include <miopen/pdist/utils.hpp>
+#include <../test/ford.hpp>
+
+template <typename Tgpu, typename Tcheck>
+int32_t mloPdistBackwardRunHost(const miopenTensorDescriptor_t inputDesc,
+                                const Tgpu* input,
+                                const Tgpu* output,
+                                const Tgpu* doutput,
+                                Tcheck* dinputHost,
+                                const double p)
+{
+    auto input_numel = miopen::deref(inputDesc).GetElementSize();
+    auto N           = miopen::deref(inputDesc).GetLengths()[0];
+    auto M           = miopen::deref(inputDesc).GetLengths()[1];
+
+    // Fill dinputHost with zeros
+    for(size_t i = 0; i < input_numel; i++)
+    {
+        dinputHost[i] = 0;
+    }
+
+    for(int i = 0; i < N; ++i)
+    {
+        for(int j = i + 1; j < N; ++j)
```

Comment:
What's the reason to use `uint64_t`? If it was about the semantic problem, then I think using `size_t` seem to be more proper?

Reply:
I don't think we can use par_ford() here, because I need to accumulate the results later. If we use parallel computing, the accumulation would require atomic operations, which would complicate things:

```
dinputHost[i * M + m] += res;
dinputHost[j * M + m] -= res;
``` 

An alternative would be to use parallel computing by creating a temporary intermediate variable to store the partial sums (Calculate this partial sum can use parallel computatuon). And finally,  do a sumReduction step to compute the final result—similar to the approach used in the kernel code. However, I don’t think this is necessary since the purpose of this function is to verify correctness rather than focus on performance.

If you notice any opportunities for parallelization in this code, please point them out specifically. At the moment, I don’t see how it could be computed in parallel. Thanks.

Reply:
N is size_t (uint64_t) while i,j is still int which leads to failed CI/CD test. use size_t is fine